### ISSUE TITLE
ERROR: create_snapshot: HTTP POST FAILURE

### ISSUE THREAD

**Issue by talisto** [alestic/ec2-consistent-snapshot#54]
I'm getting this error when I attempt to run this script:

```
[root@localhost bin]# ./ec2-consistent-snapshot -d --region us-west-2 vol-2618ff34
ec2-consistent-snapshot: Using AWS access key: <removed>
ec2-consistent-snapshot: Using default snapshot description
ec2-consistent-snapshot: Using description 'ec2-consistent-snapshot' for all snapshot descriptions
ec2-consistent-snapshot: Sat Feb 28 04:41:05 2015: sync
ec2-consistent-snapshot: Sat Feb 28 04:41:05 2015: create EC2 object
ec2-consistent-snapshot: Endpoint: https://ec2.us-west-2.amazonaws.com
ec2-consistent-snapshot: volume_id: vol-2618ff34; description: ec2-consistent-snapshot
ec2-consistent-snapshot: Sat Feb 28 04:41:05 2015: ec2-create-snapshot vol-2618ff34
ec2-consistent-snapshot: ERROR: create_snapshot: HTTP POST FAILURE: 500 Can't connect to ec2.us-west-2.amazonaws.com:443 at ./ec2-consistent-snapshot line 295.
ec2-consistent-snapshot: Sat Feb 28 04:41:05 2015: done
```

I'm not really sure how to go about debugging this.  I'm able to make snapshots just fine using the `aws` command, and I can access `ec2.us-west-2.amazonaws.com:443` just fine from that host using other programs (telnet, etc).  I'm wondering if this is related to issue #47, but I'm not sure how to proceed from here.

Any advice on how to debug this further would be much appreciated.

**Comment by ehammond** [comment]
I am unable to reproduce this issue in us-west-2.

**Comment by talisto** [comment]
I've found what was causing this problem, and it has nothing to do with your script, so I apologize for the bug report, and I appreciate that you attempted to reproduce it.  I'll explain the issue here, as someone else is bound to run into the same issue.

The problem is with the latest release of the `Mozilla::CA` perl module.  In release 20141217, the certificate "Verisign Class 3 Public Primary Certification Authority" was removed (see bug report here: https://rt.cpan.org/Public/Bug/Display.html?id=101908).  But that appears to be the certificate that Amazon is currently using:

```
---
Server certificate
subject=/C=US/ST=Washington/L=Seattle/O=Amazon.com Inc./CN=ec2.us-west-2.amazonaws.com
issuer=/C=US/O=VeriSign, Inc./OU=VeriSign Trust Network/OU=Terms of use at https://www.verisign.com/rpa (c)10/CN=VeriSign Class 3 Secure Server CA - G3
---
```

So I've found two workarounds for this issue.  One, is to downgrade `Mozilla::CA`:

```
cpan ABH/Mozilla-CA-20130114.tar.gz
```

The other workaround is to set an environment variable to specify a different certificate bundle:

```
export HTTPS_CA_FILE=/etc/ssl/certs/ca-bundle.crt
```

However, specifying a CA file seems to skip hostname checks entirely, which seems like a bad idea, so another variable also needs to be set to force hostname checking:

```
export PERL_LWP_SSL_VERIFY_HOSTNAME=1
```

I've tested both of these solutions on CentOS 7.
